src/lib.rs

```diff
@@ -99,22 +103,57 @@ impl fmt::Display for Repository {
 
 pub fn collect_stars(config: Config) -> Result<(), Box<dyn error::Error>> {
     let mut builder = ClientBuilder::new();
+    let mut print_auth_warning = false; 
 
     if let Some(ref token) = config.token {
         builder.set_authorization_token(token.to_owned());
     }
+    else {
+        print_auth_warning = true; 
+    }
 
     let client = builder.build()?;
 
     let mut stars: Vec<Star> = Vec::new();
 
     let mut next_link = config.url();
+    
+    let mut remaining: i32 = 0; 
+    let mut total: i32 = 0;
+    let mut reset_time = SystemTime::now();
 
     while next_link.is_some() {
         if let Some(link) = next_link {
             let mut res = client.get(&link).send()?;
-            next_link = extract_link_next(res.headers());
 
+            for header in res.headers().iter() {
+                match header.name() {
+                    "X-RateLimit-Limit" => {
+                        total = header.value_string().parse::<i32>()?;
+                    },
+                    "X-RateLimit-Remaining" => { 
+                        remaining = header.value_string().parse::<i32>()?;
+                    },
+                    "X-RateLimit-Reset" => {
+                        let seconds = header.value_string().parse::<u64>()?;
+
+                        // Creates a new SystemTime from the specified number of whole seconds
+                        reset_time = UNIX_EPOCH + Duration::from_secs(seconds);
```

Comment:
I think this got lost in the last review, so I'm reposting it:

My understanding from [GitHub's API docs](https://developer.github.com/v3/#rate-limiting) was that `X-RateLimit-Reset` is already a value in UTC epoch seconds. Couldn't we create a `chrono::DateTime` directly from this value?